### PR TITLE
feature/components-creation

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: frontend/src/components/TodoList.tsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import React from "react";
+import { Todo } from "../models/todo.model";
+
+import TodoListItem from "./TodoListItem";
+
```
Comment (ZunairaT) [inline frontend/src/components/TodoList.tsx:5]: improve formatting by grouping relative imports together, the space in between is unnecessary

File: frontend/src/components/TodoList.tsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+export default function TodoList(props: any) {
+  console.log(props.todos);
+  return (
```
Comment (ZunairaT) [inline frontend/src/components/TodoList.tsx:11]: remove unnecessary console log

File: frontend/src/components/TodoList.tsx [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import React from "react";
+import { Todo } from "../models/todo.model";
+
+import TodoListItem from "./TodoListItem";
+
+export default function TodoList(props: any) {
+  console.log(props.todos);
```
Comment (ZunairaT) [inline frontend/src/components/TodoList.tsx]: Do you know about the expected type of props? if yes then replace any with a model or type

File: frontend/src/App.tsx [hunk 4]
```diff
@@ -1,24 +1,20 @@
...
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
+      <div className="card">
+      <Input /> <button id="bt">Post</button>
```
Comment (ZunairaT) [inline frontend/src/App.tsx]: ids and classes should be name in a better way so its understandable `bt` is very random. You should follow bem naming convention https://www.freecodecamp.org/news/css-naming-conventions-that-will-save-you-hours-of-debugging-35cea737d849/

**Review by ZunairaT — CHANGES_REQUESTED** [review]

File: frontend/src/components/Input.tsx [hunk 5]
```diff
@@ -0,0 +1,5 @@
+import React from "react";
+
+export default function Input(props: any) {
+  return <input placeholder={props.sample} onChange={props.changeValue} />;
+}
```
Comment (alinauman02) [inline frontend/src/components/Input.tsx:32]: This component is supposed to be reusable. It should be possible to use this for different kinds of data.

To make a component reusable, you pass it a number of `props`. Take a look [here](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#attributes) to see the attributes you can add to an `input` element and see if you can identify which one of these could be passed as `props`.

Comment (alinauman02) [inline frontend/src/components/Input.tsx:32]: Also, you should use appropriate names for all variables. `props.sample` is not very clear - what does sample mean? Which attribute do you intend to bind it to? `props.placeholder` is much more clear. Also, `props.changeValue` should be `props.onChange`.

File: frontend/src/components/TodoList.tsx [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+      <table>
+        {props.todos.map((todo: Todo) => (
+          <tr>
+            <TodoListItem todo={todo} />
+          </tr>
+        ))}
+      </table>
+    </div>
+  );
+}
```
Comment (alinauman02) [inline frontend/src/components/TodoList.tsx:27]: As discussed on Friday, the `table` element is not appropriate to use in this scenario. The UI given does not have a table in it. Replace `table` with a more semantically appropriate element, like `ul`.

File: frontend/src/components/TodoListItem.tsx [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+    <tr>
+      <td>
+        <p>{props.todo.description}</p>
+      </td>
+      <td>
+        <button>Delete</button>
+      </td>
+    </tr>
+  );
+}
```
Comment (alinauman02) [inline frontend/src/components/TodoListItem.tsx:19]: Like discussed earlier, this component should return a `li`, which contains the text and the delete button.

File: frontend/src/App.tsx [hunk 8]
```diff
@@ -1,9 +1,20 @@
 import "./App.css";
 
+import "./App.css";
+import Input from "./components/Input";
+import TodoList from "./components/TodoList";
+import { Todo } from "./models/todo.model";
+let todos: Todo[] = [];
```
Comment (alinauman02) [inline frontend/src/App.tsx]: This list of todo should be the `App` component's state, as any change in this should cause the UI to update. Move this inside the `App` component function and declare it using `useState`.

**Review by alinauman02 — CHANGES_REQUESTED** [review]
Please address the changes I have requested

File: frontend/src/App.css [hunk 9]
```diff
@@ -3,3 +3,70 @@
...
+}
+input:hover {
+  border-color: #24a0ed;
+}
+input {
+  width: 100%;
+  padding: 3%;
+  margin: 1%;
+}
+ul {
```
Comment (alinauman02) [inline frontend/src/App.css]: You are still using element selectors like `input` and `ul`. This means that all the input fields and all unordered lists will have the same style. We do not generally want to use the same style for all instances of an element.

Replace these with a class and add it to the Todo input and list elements.

File: frontend/src/App.css [hunk 10]
```diff
@@ -3,3 +3,70 @@
...
+  display: inline-block;
+}
+.del-icon {
+  width: 70%;
+  height: 70%;
+}
+.delete-button {
+  padding: 2%;
+  background-color: #0a2ea4;
+}
```
Comment (alinauman02) [inline frontend/src/App.css:86]: Also, add an empty line after each `}` to make the file easier to read.

File: frontend/src/App.tsx [hunk 11]
```diff
@@ -1,9 +1,38 @@
...
+import { Todo } from "./models/todo.model";
 
 function App() {
+  const [todos, setTodos] = useState<Todo[]>([
+    { description: "Hello world", id: "244454", isDone: false },
+    { description: "Hello Pakistan", id: "24454", isDone: false },
+    { description: "Hello GB", id: "24433", isDone: false },
+  ]);
+
+  let value: string = "";
```
Comment (alinauman02) [inline frontend/src/App.tsx]: `value` is a vague name. `newTodo` is more appropriate and tells that this holds the description for a new Todo. Also, this should be a state variable, not a regular `let`/`const` variable, as we want the component to re-render as this changes.

File: frontend/src/App.tsx [hunk 12]
```diff
@@ -1,9 +1,38 @@
...
   return (
-    <div>
-      <p>Hello world!</p>
+    <div className="App">
+      <div className=" card">
+        <div className="flex-container">
+          <Input
+            onChange={onChange}
+            value={value}
+            placeholder="ENTER TASK"
```
Comment (alinauman02) [inline frontend/src/App.tsx]: The `placeholder` attribute takes a value that is an example of the data a user can enter in the input field. It's the label that tells the name of the field.

For example, if I am taking an input for an email address, the placeholder I would use is `someone@example.com`.

Update the placeholder to something like "Learn React" (without the quotes).

File: frontend/src/components/Input.tsx [hunk 13]
```diff
@@ -0,0 +1,34 @@
+import React, { ChangeEvent } from "react";
+import { useState } from "react";
+interface InputProps {
```
Comment (alinauman02) [inline frontend/src/components/Input.tsx:3]: Add an empty line between the imports and the `InputProps` interface. Also, replace both the imports with this line: 

`import { ChangeEvent, useState } from "react";`

File: frontend/src/components/TodoList.tsx [hunk 14]
```diff
@@ -0,0 +1,16 @@
+import React from "react";
+import { Todo } from "../models/todo.model";
+
+import TodoListItem from "./TodoListItem";
+
+export default function TodoList(props: any) {
```
Comment (alinauman02) [inline frontend/src/components/TodoList.tsx]: Assign a proper type to `props`

File: frontend/src/components/TodoListItem.tsx [hunk 15]
```diff
@@ -0,0 +1,13 @@
+import { IconDelete } from "./Icons/IconDelete";
+export default function TodoListItem(props: any) {
```
Comment (alinauman02) [inline frontend/src/components/TodoListItem.tsx]: Assign a proper type to `props` here as well. Also add an empty line after the `IconDelete` import statement.

File: frontend/src/App.css [hunk 16]
```diff
@@ -3,3 +3,84 @@
...
+
+.card {
+  padding: 2% 3%;
+  width: 30%;
+  background-color: white;
+  border-radius: 5%;
+}
+
+input:hover {
```
Comment (alinauman02) [inline frontend/src/App.css]: This is still using an element selector. Change it to `input-todo:hover`.

File: frontend/src/components/TodoList.tsx [hunk 17]
```diff
@@ -2,13 +2,20 @@ import React from "react";
 import { Todo } from "../models/todo.model";
 
 import TodoListItem from "./TodoListItem";
-
-export default function TodoList(props: any) {
+interface todoList {
```
Comment (alinauman02) [inline frontend/src/components/TodoList.tsx]: Interfaces that define the shape of a components props are usually named like this: ComponentName + Props

e.g. We used ` InputProps` for the `Input` component.  Change this to `TodoListProps`. Also use destructuring to access todos.

File: frontend/src/components/TodoListItem.tsx [hunk 18]
```diff
@@ -1,9 +1,10 @@
+import { Todo } from "../models/todo.model";
 import { IconDelete } from "./Icons/IconDelete";
-export default function TodoListItem(props: any) {
+export default function TodoListItem(props: Todo) {
```
Comment (alinauman02) [inline frontend/src/components/TodoListItem.tsx]: You always create a new interface to define the props of a component. That makes it easier to add/remove props.

Create an interface named `TodoListItemProps` and put `todo: Todo` inside it. Also use destructuring here.

**Review by alinauman02 — APPROVED** [review]

**Review by ZunairaT — APPROVED** [review]

**Pull request merged**